### PR TITLE
Util.h extensions

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Add UTIL_OR and UTIL_AND, along with tests for MACRO_MAP_CAT.

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by pabigot — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: include/sys/util.h [hunk 1]
```diff
@@ -711,6 +711,21 @@ u8_t u8_to_dec(char *buf, u8_t buflen, u8_t value);
...
+ *
+ * UTIL_OR(foo, bar) is sometimes a replacement for (foo || bar)
+ * when "bar" is an expression that would cause a build
+ * error when "foo" is true.
+ *
+ * UTIL_AND(foo, bar) is sometimes a replacement for (foo && bar)
+ * when "bar" is an expression that would cause a build
+ * error when "foo" is false.
+ */
+#define UTIL_OR(a, b) COND_CODE_1(UTIL_BOOL(a), (a), (b))
```
Comment (nordic-krch) [inline include/sys/util.h]: shouldn't it be ` COND_CODE_1(UTIL_BOOL(a), UTIL_BOOL(a),UTIL_BOOL(b))`?

File: include/sys/util.h [hunk 2]
```diff
@@ -711,6 +711,21 @@ u8_t u8_to_dec(char *buf, u8_t buflen, u8_t value);
...
+#define UTIL_AND(a, b) COND_CODE_1(UTIL_BOOL(a), (b), (0))
```
Comment (nordic-krch) [inline include/sys/util.h]: same here

**Review by nordic-krch — COMMENTED** [review]
I'm late with the review but really cool macros! I was not aware of `UTIL_BOOL` and i missed that in few places. Will probably get back with something like `COND_CODE_NZ`

**Comment by nordic-krch** [comment]
> I was not aware of UTIL_BOOL and i missed that in few places.

Got excited too early. `UTIL_BOOL` returns true when define does not exist so it does not solve the problem I wanted to solve.

File: include/sys/util.h (hunk 1 above)
Comment (mbolivar-nordic) [reply include/sys/util.h]: I wanted them to return the actual nonzero value for things like this:

```
#define DT_NUM_INST(compat)					\
	UTIL_AND(DT_HAS_COMPAT(compat),				\
		 UTIL_CAT(DT_N_INST, DT_DASH(compat, NUM)))
```

Think it's worth changing?

**Pull request merged**